### DISCUSSION TITLE
Subflows + Grouping + Auto Layout

### DISCUSSION THREAD

**Discussion by fum4** [xyflow/xyflow#4809] [Layouting]
Hello,

I'm having a hard time making subflows, grouping and auto layouting work together. I think I get the big picture, but after so many failed attempts I'm not that sure anymore.

I created a demo app ([repo](https://github.com/fum4/reactflow-subflows-autolayout) + [live demo](https://reactflow-subflows-autolayout.vercel.app/)) including part of my logic, the relevant part. It uses react-dnd and d3 for drag'n'drop and layout respectively. It is stripped down of my failed attempts at grouping the nodes because it would only complicate the issue as for sure I wasn't going in the right direction.

Hopefully everything is properly documented in the [README](https://github.com/fum4/reactflow-subflows-autolayout/blob/master/README.md).

Any help is highly appreciated! Hopefully there is already an existing discussion / issue that I didn't stumble upon that would shed some light on this. If so, sorry for that :) 

Thank you!

**Comment by elenamik** [comment]
I am also struggling to find a solution that supports auto layouting with grouped nodes. I am thinking the best option is to use the subflows feature and the elk layouting library suggested in the docs?

**Reply by shahsagarm** [reply]
have you figured this out?
